src/uthash.h

```diff
@@ -825,44 +898,45 @@ do {
   _he_new_buckets = (UT_hash_bucket*)uthash_malloc(                              \
            2UL * tbl->num_buckets * sizeof(struct UT_hash_bucket));              \
   if (!_he_new_buckets) {                                                        \
-    uthash_fatal("out of memory");                                               \
-  }                                                                              \
-  uthash_bzero(_he_new_buckets,                                                  \
-          2UL * tbl->num_buckets * sizeof(struct UT_hash_bucket));               \
-  tbl->ideal_chain_maxlen =                                                      \
-     (tbl->num_items >> (tbl->log2_num_buckets+1U)) +                            \
-     (((tbl->num_items & ((tbl->num_buckets*2U)-1U)) != 0U) ? 1U : 0U);          \
-  tbl->nonideal_items = 0;                                                       \
-  for (_he_bkt_i = 0; _he_bkt_i < tbl->num_buckets; _he_bkt_i++) {               \
-    _he_thh = tbl->buckets[ _he_bkt_i ].hh_head;                                 \
-    while (_he_thh != NULL) {                                                    \
-      _he_hh_nxt = _he_thh->hh_next;                                             \
-      HASH_TO_BKT(_he_thh->hashv, tbl->num_buckets * 2U, _he_bkt);               \
-      _he_newbkt = &(_he_new_buckets[_he_bkt]);                                  \
-      if (++(_he_newbkt->count) > tbl->ideal_chain_maxlen) {                     \
-        tbl->nonideal_items++;                                                   \
-        _he_newbkt->expand_mult = _he_newbkt->count / tbl->ideal_chain_maxlen;   \
-      }                                                                          \
-      _he_thh->hh_prev = NULL;                                                   \
-      _he_thh->hh_next = _he_newbkt->hh_head;                                    \
-      if (_he_newbkt->hh_head != NULL) {                                         \
-        _he_newbkt->hh_head->hh_prev = _he_thh;                                  \
+    HASH_MEM_FAILED(mem_ok);                                                     \
+  } else {                                                                       \
+    uthash_bzero(_he_new_buckets,                                                \
+        2UL * tbl->num_buckets * sizeof(struct UT_hash_bucket));                 \
```

Comment:
Hygiene nit: This should really be `(tbl)` throughout. (If you submit this as a separate PR, I'll merge it.)